chain-abstraction/sol-implementation/types.go

```diff
@@ -60,23 +76,46 @@ func (a *Assertion) inner() (*rollupgen.AssertionNode, error) {
 			a.id,
 		)
 	}
+	// Update the assertion with the latest data, if they are in now in constant state.
+	if assertionNode.FirstChildBlock > 0 {
+		a.firstChildBlock = option.Some(assertionNode.FirstChildBlock)
+	}
+	if assertionNode.SecondChildBlock > 0 {
+		a.secondChildBlock = option.Some(assertionNode.SecondChildBlock)
+	}
+	if assertionNode.IsFirstChild {
+		a.isFirstChild = option.Some(true)
+	}
+	assertionStatus := protocol.AssertionStatus(assertionNode.Status)
+	if assertionStatus != protocol.AssertionConfirmed {
```

Comment:
we should only cache if the status is confirmed here, otherwise, it could be changing

Reply:
Maybe here just do the same as with edges, add a property called isConfirmed and when we query for assertion status, return confirmed if true

Reply:
done.